src/book/03-guides/01-main-codebase/07-onboarding-task.mdx

```diff
@@ -0,0 +1,486 @@
+
+    - `on<Configuration>` is used to load in configuration values from file `Pong.yaml`
+    - `on<Startup>()` runs just once when the module starts
+    - `on<Trigger<Ping>>()` runs when `Ping` message is received and prints "Ping"
+
+<Alert type='info'>
+
+The `log<NUClear::INFO>("Ping");` can be thought of as the equivalent of an `std::cout` in normal C++ or a `print` in python. There are various types that can be used for logging information. See NUbook for more information.
+
+</Alert>
+
+## Task: Git Branching
+
+A branch is your own workspace where you can make changes and implement new features. It is a good practice to create a new branch for each new feature you are working on. This way, you can work on multiple features at the same time without affecting the main codebase.
+
+1.  Open a terminal in your `NUbots` directory and checkout the branch `montano/onboarding` by running
+
+    ```bash
+    git checkout montano/onboarding
+    ```
+
+    This branch has a role `onboarding.role` and module `Ping` already created for you.
+
+2.  Now crate your own branch which extends from this by running
+
+    ```bash
+    git checkout -b <your-last-name>/branching-task
+    ```
+
+<Alert type='info'>
+
+Don't be afraid to break things in your own branch! It will not directly impact the `main` branch.
+
+</Alert>
+
+### VSCode Git Extension
+
+There exists many GUI tools for using Git which can make your life easier, for example VSCode, Github Desktop or Git Kraken.
+
+In VScode, you can create your own branch by clicking on the button on the bottom left hand side, which is the name of the branch that you are currently on.
+
+To create a new branch using the VSCode Git extension:
+
+1.  Click on the button in the bottom left hand corner of the screen containing the name of the current branch.
+
+    If you have correctly done the git checkout command, then you should be in the `montano/onboarding` branch.
+
+2.  Select `Create new branch from...` and enter `montano/onboarding`. This will create a new branch starting from the `montano/onboarding` branch.
+
+    ![Create new branch from](images/onboarding_task/branching/cnbf.png)
+
+3.  Enter a branch name which should follow convention `<your-last-name>/<branch-name>`, where branch name should be something clear and simple.
+
+## Task: Creating messages and modules
+
+You should see the `Ping` module already implemented in `NUbots/module/onboarding`.
+
+Currently, this code **will not compile** because a `Pong` message doesn't exist, so let's create it.
+
+1. Navigate to `NUbots/shared/message/onboarding`, duplicate `Ping.proto` and rename it and its contents to `Pong`.
+
+   You should have a new file `Pong.proto` in folder `NUbots/shared/message/onboarding` with contents:
+
+   ```protobuf
+   syntax = "proto3";
+
+   package message.onboarding;
+
+   message Pong{
+   }
+   ```
+
+2. Ensure that the code now compiles by running:
+
+   ```bash
+   ./b configure --clean
+   ./b build
+   ```
```

Comment:
Might be good to talk about `-i` and turning off other roles, some of the new peeps have slow laptops and struggled to build during the first recruitment task